csi/volumeutils.py

```diff
@@ -1009,13 +1015,224 @@ def reload_glusterfs(volume):
     return True
 
 
-def mount_glusterfs(volume, mountpoint, is_client=False):
+class VolfileElement:
+    def __init__(self, name):
+        self.name = name
+        self.type = None
+        self.options = {}
+        self.subvolumes = None
+
+
+class Volfile:
+    def __init__(self, volfile, elements):
+        self.volfile = volfile
+        self.elements = elements
+
+    @classmethod
+    def parse(cls, volfile):
+        element = None
+        elements = []
+        with open(volfile) as volf:
+            for line in volf:
+                line = line.strip()
+                if line.startswith("volume "):
+                    element = VolfileElement(line.split()[-1])
+                    #print(line.split()[-1])
+                elif line == "end-volume":
+                    elements.append(element)
+                    element = None
+                elif line.startswith("option "):
+                    _, name, value = line.split()
+                    element.options[name] = value
+                elif line.startswith("subvolumes "):
+                    element.subvolumes = line.split(" ", 1)[-1]
+                elif line.startswith("type "):
+                    element.type = line.split()[-1]
+                    # print("type")
+                    # print(line.split()[-1])
+        return Volfile(volfile, elements)
+
+    def update_options_by_type1(self, opts):
+        for element in self.elements:
+            if opts.get(element.type, None) is not None:
+                logging.info(logf(
+                    "opts[element.type]",
+                    opts_element_type=opts[element.type],
+                    element_type=element.type
+                ))
+                element.options.update(opts[element.type])
+                logging.info(logf(
+                    "opts[element.type]",
+                    opts_element_type=opts[element.type],
+                    element_type=element.type
+                ))
+
+    def update_options_by_type2(self, opts):
+        opt = {}
+        for element in self.elements:
+
+            # Example: 'cluster/replicate' & 'performance/open-behind'
```

Comment:
You can't hard code performance here. There are many xlators where option name is same as xlator name to switch off/on. For example: `features/changelog.changelog=on`

For now accept Volume options as `performance/open-behind.open-behind: on`. Introduce alias support then we can have a json mapping for Option name and expanded option names. 

All supported options(Also helps us to validate)

```json
{
    "performance.open-behind": "performance/open-behind.open-behind"
}
```

Reply:
Alias support can be in next PR. This PR can accept fully expanded Option names only.

Reply:
done